opensaas generate fails with "Cannot find module" when tsconfig.json has a bare "*" path (e.g. { "*": ["./src/*"] })

## Summary

`opensaas generate` fails to load `opensaas.config.ts` when the project's `tsconfig.json` has a bare `"*"` catch-all path pattern — a common convention for unprefixed imports like `lib/utils` (e.g. shadcn/ui-style setups):

```json
{
  "compilerOptions": {
    "paths": {
      "*": ["./src/*"],
      "@/*": ["./src/*"]
    }
  }
}
```

### Error

```
❌ Error: Cannot find module '/abs/path/to/project/src/abs/path/to/project/opensaas.config.ts'
Require stack:
- /abs/path/to/project/_index.js
```

## Root cause

`resolveTsconfigAlias` (`packages/cli/src/generator/tsconfig-alias.ts`, added in #932 / 0.39.0) translates `tsconfig.json` `paths` into `jiti`'s `alias` option by stripping the trailing `*` from both the pattern and target. For the bare pattern `"*"`, stripping the trailing `*` leaves an **empty-string alias key**.

`jiti` delegates alias resolution to `pathe`'s `resolveAlias`, which matches by prefix (`id.startsWith(key)`). Every string starts with `''`, so the empty key ends up aliasing **every module specifier** jiti resolves — not just the unprefixed imports the `"*"` pattern was meant to cover. This corrupts resolution of `opensaas.config.ts`'s own absolute path and its entire import closure (including plain relative imports that never used the alias), causing `opensaas generate` to fail outright for any project using this otherwise-common tsconfig pattern.

## Fix

I've already pushed a fix + regression tests + changeset to a branch in this repo: `claude/opensaas-upgrade-0-39-0-xrfakk`. It skips (and warns on) a pattern whose alias key resolves to the empty string, the same way other unrepresentable `paths` entries are already skipped — `packages/cli/src/generator/tsconfig-alias.ts`.

Filed per this project's own `CLAUDE.md` guidance to log stack issues discovered while upgrading a downstream app (`borisno2/emily-calder-performing-arts`, upgrading to 0.39.0). Happy to open a PR from that branch if useful — let me know.

## Repro

```bash
mkdir repro && cd repro
cat > tsconfig.json <<'EOF'
{ "compilerOptions": { "paths": { "*": ["./src/*"] } } }
EOF
echo 'export default { lists: {} }' > opensaas.config.ts
npx @opensaas/stack-cli@0.39.0 generate
```

## Version

`@opensaas/stack-cli@0.39.0` (introduced by #932, the tsconfig path-alias resolution feature)

--- a/.changeset/tame-suns-relax.md
+++ b/.changeset/tame-suns-relax.md
@@ -1,0 +1,5 @@
+---
+'@opensaas/stack-cli': patch
+---
+
+Fix `resolveTsconfigAlias` corrupting resolution of `opensaas.config.ts` and its whole import closure when `tsconfig.json` has a bare `"*"` path pattern (e.g. `{ "*": ["./src/*"] }`, a common catch-all for unprefixed imports like `lib/utils`). The bare pattern now produces an empty alias key, which jiti's prefix-based resolution would otherwise match against every specifier; it is now skipped and reported as a warning like other unrepresentable path entries.

--- a/packages/cli/src/generator/tsconfig-alias.test.ts
+++ b/packages/cli/src/generator/tsconfig-alias.test.ts
@@ -106,6 +106,19 @@ describe('resolveTsconfigAlias', () => {
     expect(warnings).toHaveLength(1)
   })
 
+  it('warns and skips a bare "*" catch-all pattern instead of aliasing every specifier', () => {
+    writeTsconfig({
+      compilerOptions: { paths: { '*': ['./src/*'], '@utils/*': ['./src/utils/*'] } },
+    })
+
+    const { alias, warnings } = resolveTsconfigAlias(tempDir)
+
+    expect(alias).toEqual({ '@utils/': path.join(tempDir, 'src', 'utils') })
+    expect(warnings).toHaveLength(1)
+    expect(warnings[0]).toContain('"*"')
+    expect(warnings[0]).toContain('catch-all')
+  })
+
   it('mixes representable and unrepresentable entries: resolves one, warns on the other', () => {
     writeTsconfig({
       compilerOptions: {
@@ -213,6 +226,23 @@ describe('resolveTsconfigAlias + jiti integration', () => {
       default: { lists: { hello: string } }
     }
     expect(mod.default.lists).toEqual({ hello: 'transitive' })
+  })
+
+  it('a bare "*" catch-all does not corrupt resolution of the config’s own absolute path (issue repro)', async () => {
+    fs.writeFileSync(
+      path.join(tempDir, 'tsconfig.json'),
+      JSON.stringify({ compilerOptions: { paths: { '*': ['./src/*'] } } }),
+    )
+    fs.writeFileSync(path.join(tempDir, 'opensaas.config.ts'), 'export default { lists: {} }\n')
+
+    const { alias } = resolveTsconfigAlias(tempDir)
+    expect(alias).toBeUndefined()
+
+    const jiti = createJiti(tempDir, { interopDefault: true, alias })
+    const mod = (await jiti.import(path.join(tempDir, 'opensaas.config.ts'))) as {
+      default: { lists: Record<string, never> }
+    }
+    expect(mod.default.lists).toEqual({})
   })
 
   it('fails to resolve the alias when no tsconfig.json is present (baseline without the fix)', async () => {

--- a/packages/cli/src/generator/tsconfig-alias.ts
+++ b/packages/cli/src/generator/tsconfig-alias.ts
@@ -35,6 +35,16 @@ export interface TsconfigAliasResult {
  * (`extends` chains, project references, conditional exports) is out of
  * scope.
  *
+ * A bare `"*"` pattern (`{ "*": ["./src/*"] }`, a common catch-all for
+ * unprefixed imports like `lib/utils`) is a single-trailing-star pattern too,
+ * but stripping its trailing `*` leaves an EMPTY alias key. `pathe`'s
+ * `resolveAlias` matches keys by prefix (`id.startsWith(key)`), and every
+ * string starts with `''` — so an empty key aliases every module specifier
+ * jiti resolves, including `opensaas.config.ts`'s own absolute path and every
+ * relative import in its closure, not just the bare imports it was meant for.
+ * This is therefore skipped and reported as a warning like any other
+ * unrepresentable entry, rather than corrupting unrelated resolution.
+ *
  * A project with no `tsconfig.json`, or one with no `paths`, returns
  * `{ alias: undefined, warnings: [] }` so callers behave exactly as they did
  * before this translation existed.
@@ -87,6 +97,13 @@ export function resolveTsconfigAlias(cwd: string): TsconfigAliasResult {
     }
 
     const aliasKey = pattern.slice(0, -1)
+    if (aliasKey === '') {
+      warnings.push(
+        `tsconfig.json path "${pattern}" -> "${target}" is a catch-all alias (empty prefix); jiti's prefix-based alias resolution would apply it to every module specifier, not just unprefixed imports, so this alias will not be resolved.`,
+      )
+      continue
+    }
+
     const aliasTarget = path.resolve(baseDir, target.slice(0, -1))
     alias[aliasKey] = aliasTarget
   }

